### PR TITLE
ReleaseNotes91CU4

### PR DESCRIPTION
**Pull request by DivyaC0403**

### CODE REVIEW

File: release_notes/Service_Fabric_ReleaseNotes_90CU9.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+|Java SDK  |Java for Linux SDK  | 1.0.6 |
+|Service Fabric PowerShell and CLI | AzureRM PowerShell Module  <br> SFCTL |  0.3.15  <br> 11.0.1 |
+
+## Service Fabric Feature and Bug Fixes
+| Versions | IssueType | Description | Resolution | 
+|-|-|-|-|
+| **Windows - 9.0.1526.9590<br>Ubuntu 18 - 9.0.1463.1<br>Ubuntu 20 - 9.0.1463.1** | **Feature** | Key Value Store (KVS) | **Brief Description**:  KVS backed partitions, including Actors and System Services, experience quorum loss due to StoreLVIDLimitHit error code (0x80071d4d), indicating the depletion of values for a monotonically increasing ID field. This results in unavailability of the partition until a manual offline maintenance task is executed to resolve the issue. <br>**Solution**: To address the error, KVS replicas now restart automatically and initiate the maintenance task (Extensible Storage Engine database compaction) during the next open. While there may be temporary availability loss in partitions depending on the task duration, no manual intervention is required to mitigate the issue.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_90CU9.md:25]: While there may be temporary availability loss in partitions depending on the task duration, no manual intervention is required to mitigate the issue.

This feels very open ended, can we quantify with any particular typical experience that is expected from testing this?

File: release_notes/Service_Fabric_ReleaseNotes_90CU9.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+* Ubuntu 18.04 LTS will reach its 5-year end-of-life window on June, 2023. Service Fabric runtime will drop support for 18.04 LTS after the published date, and we recommend moving your clusters and applications to supported versions listed here: [Service Fabric supported Linux versions](https://learn.microsoft.com/en-us/azure/service-fabric/service-fabric-versions#supported-linux-versions-and-support-end-date)
+* Service Fabric runtime will soon stop supporting BinaryFormatter based remoting exception serialization by default and move to using Data Contract Serialization (DCS) based remoting exception serialization by default. Current applications using it will continue to work as-is, but Service Fabric strongly recommends moving to using Data Contract Serialization (DCS) based remoting exception instead.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_90CU9.md:31]: Service Fabric runtime will soon stop supporting BinaryFormatter

Are we actually dropping support for BinaryFormatter if customers use it or just modifying the default?

File: release_notes/Service_Fabric_ReleaseNotes_90CU9.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* Service Fabric runtime will soon be archiving and removing Service Fabric runtime version 6.4 packages and older, as well as SDK version 3.3 packages and older from the package Download Center. Archiving/Removing will affect application scaling and re-imaging of virtual machines in a Service Fabric Cluster running on unsupported versions. After older versions are removed/archived, this may cause failure while rolling back when the current in-progress upgrade has errors.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_90CU9.md:32]: just to be clear, we are leaving github repository with this code base available though? (seems kind of weird)

File: release_notes/Service_Fabric_ReleaseNotes_90CU9.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+| **Windows - 9.0.1526.9590<br>Ubuntu 18 - 9.0.1463.1<br>Ubuntu 20 - 9.0.1463.1** | **Feature** | Key Value Store (KVS) | **Brief Description**: Auto-compaction now emits health events at the start and completion, visible in the Service Fabric Explorer, enhancing its monitoring capabilities. These events complement trace data, providing effective visibility into active auto-compaction progress.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_90CU9.md:26]: Can we link to SFX repo/show an example of where customers can find this data?

File: release_notes/Service_Fabric_ReleaseNotes_91CU4.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+| Service | Platform | Version |
+|---------|----------|---------|
+|Service Fabric Runtime| Ubuntu 18 <br> Ubuntu 20 <br> Windows | 9.1.1592.1 <br> 9.1.1592.1 <br> 9.1.1799.9590 |
+|Service Fabric for Windows Server|Service Fabric Standalone Installer Package | 9.1.1799.9590 |
+|.NET SDK |Windows .NET SDK <br> Microsoft.ServiceFabric <br> Reliable Services and Reliable Actors <br> ASP.NET Core Service Fabric integration| 6.1.1799  <br> 9.1.1799 <br> 6.1.1799 <br> 6.1.1799 |
+|Java SDK  |Java for Linux SDK  | 1.0.6 |
+|Service Fabric PowerShell and CLI | AzureRM PowerShell Module  <br> SFCTL |  0.3.15  <br> 11.0.1 |
+
+
+## Service Fabric Feature and Bug Fixes
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_91CU4.md:20]: Similar comments above apply here

File: release_notes/Service_Fabric_ReleaseNotes_91CU4.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+| **Windows - 9.1.1799.9590<br>Ubuntu 18 - 9.1.1592.1<br>Ubuntu 20 - 9.1.1592.1** | **Feature** | Failover Manager | **Brief Description**:  When MinInstancePercentage is set to 100 and the stateless service has N instances, Service Fabric creates additional instances before taking any instances down for upgrade to ensure a total of N instances remain active throughout the upgrade process. However, during the upgrade, if FailoverManager.RelaxCheckForSafeReplicaCloseCount is set to True, only N-1 instances are guaranteed to be available. On the other hand, if FailoverManager.RelaxCheckForSafeReplicaCloseCount is set to False, the upgrade gets stuck in the preSafetyCheck phase, resulting in timeouts and rollbacks due to the unavailability of N instances <br>**Solution**: Set parameters RelaxCheckForSafeReplicaCloseCount as False and IsStrongMinInstanceCountCheckEnabled as True under FailoverManager section in ClusterManifest to move the upgrade forward.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_91CU4.md:27]: Where are the public docs for this scenario? This seems important to bubble up there and reference it?

File: release_notes/Service_Fabric_ReleaseNotes_91CU4.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+| **Windows - 9.1.1799.9590<br>Ubuntu 18 - 9.1.1592.1<br>Ubuntu 20 - 9.1.1592.1** | **Feature** | Application Model | **Brief Description**: Service Fabric has a restriction that allows only one application type with the same name to be provisioned or unprovisioned at any given time. However, if an application type contains illegal path characters in its name or version, it becomes stuck in an unprovisioning loop. This situation not only affects the problematic application type but also causes other application types with the same name to fail during provisioning. <br>**Solution**:  Improvements were made in Service Fabric to address the issue with invalid characters.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_91CU4.md:26]: Do we have docs showing what valid characters are? May be good to link to them from here.

File: release_notes/Service_Fabric_ReleaseNotes_91CU4.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+* Ubuntu 18.04 LTS will reach its 5-year end-of-life window in June, 2023. Service Fabric runtime will drop support for 18.04 LTS after the published date, and we recommend moving your clusters and applications to supported versions listed here: [Service Fabric supported Linux versions](https://learn.microsoft.com/en-us/azure/service-fabric/service-fabric-versions#supported-linux-versions-and-support-end-date)
+* Service Fabric runtime will soon stop supporting BinaryFormatter based remoting exception serialization by default and move to using Data Contract Serialization (DCS) based remoting exception serialization by default. Current applications using it will continue to work as-is, but Service Fabric strongly recommends moving to using Data Contract Serialization (DCS) based remoting exception instead.
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_91CU4.md:32]: Same comment as above

File: release_notes/Service_Fabric_ReleaseNotes_91CU4.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+* Service Fabric runtime will soon be archiving and removing Service Fabric runtime version 6.4 packages and older, as well as SDK version 3.3 packages and older from the package Download Center. Archiving/Removing will affect application scaling and re-imaging of virtual machines in a Service Fabric Cluster running on unsupported versions. After older versions are removed/archived, this may cause failure while rolling back when the current in-progress upgrade has errors.
+* Migrate Azure Active Directory Authentication Library (ADAL) library to Microsoft Authentication Library (MSAL) library, since ADAL will be out of support after December 2022. This will impact customers using AAD for authentication in Service Fabric for below features:<ul><li>Powershell, StandAlone Service Fabric Explorer(SFX), TokenValicationService</li><li>FabricBRS using AAD for keyvault authentication</li><li>KeyVaultWrapper</li><li>ms.test.winfabric.current test framework</li><li>KXM tool</li><li>AzureClusterDeployer tool</li></ul>For more information see: [MSAL Migration] (https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-migration)
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_91CU4.md:34]: What are customers expected to do? Seems there is a problem statement here but no resolution path

File: release_notes/Service_Fabric_ReleaseNotes_90CU9.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+* Service Fabric runtime will soon be archiving and removing Service Fabric runtime version 6.4 packages and older, as well as SDK version 3.3 packages and older from the package Download Center. Archiving/Removing will affect application scaling and re-imaging of virtual machines in a Service Fabric Cluster running on unsupported versions. After older versions are removed/archived, this may cause failure while rolling back when the current in-progress upgrade has errors.
+* Migrate Azure Active Directory Authentication Library (ADAL) library to Microsoft Authentication Library (MSAL) library, since ADAL will be out of support after December 2022. This will impact customers using AAD for authentication in Service Fabric for below features:<ul><li>Powershell, StandAlone Service Fabric Explorer(SFX), TokenValicationService</li><li>FabricBRS using AAD for keyvault authentication</li><li>KeyVaultWrapper</li><li>ms.test.winfabric.current test framework</li><li>KXM tool</li><li>AzureClusterDeployer tool</li></ul>For more information see: [MSAL Migration] (https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-migration)
```
Comment (craftyhouse) [inline release_notes/Service_Fabric_ReleaseNotes_90CU9.md:33]: What are customers expected to do? Seems there is a problem statement here but no resolution path

File: release_notes/Service_Fabric_ReleaseNotes_90CU9.md (hunk 2 above)
Comment (DivyaC0403) [reply release_notes/Service_Fabric_ReleaseNotes_90CU9.md:31]: We will be dropping the support for BinaryFormatter in future.

**Review by craftyhouse — APPROVED** [review]

**Pull request merged**